src/controller/CHIPDeviceController.cpp

```diff
@@ -556,6 +556,16 @@ CommissioneeDeviceProxy * DeviceCommissioner::FindCommissioneeDevice(const Trans
 
 void DeviceCommissioner::ReleaseCommissioneeDevice(CommissioneeDeviceProxy * device)
 {
+    // TODO: Call CloseSession here see #16440 and #16805 (blocking)
+
+#if CONFIG_NETWORK_LAYER_BLE
+    if (mSystemState->BleLayer() != nullptr && device->GetDeviceTransportType() == Transport::Type::kBle)
+    {
+        // We only support one BLE connection, so if this is BLE, close it
+        ChipLogProgress(Discovery, "Closing all BLE connections");
+        mSystemState->BleLayer()->CloseAllBleConnections();
```

Comment:
This is a path where it would be possible to actually close the specific connection with the connection object.  If we can, shouldn't we?  The fact we can't distinguish BLE connections by address shouldn't block our ability to do this.

Reply:
Makes sense - would you mind if I did this in a follow-up? This is a straight move of the logic from the CASEServer into the commissioner. LMK if that works for you and I can open an issue to track.

Reply:
Yes, thank you.